### PR TITLE
[ADD]sale_order_type_quotation_number: Set quotation sequences in sale order types

### PR DESCRIPTION
**Pull request by manuelregidor**
Depends on https://github.com/OCA/sale-workflow/pull/2752

This module allows to set a quotation sequence depending on the sale order type. It also makes compatible the modules sale_order_type and sale_quotation_number, so both a sale order sequence and a quotation sequence can be set in a sale order type.

### CODE REVIEW

**Comment by manuelregidor** [comment]
@ValentinVinagre @HaraldPanten

File: sale_order_type_quotation_number/views/sale_order_type_view.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2023 Manuel Regidor <manuel.regidor@sygel.es>
+ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="quotation_sot_sale_order_type_form_view" model="ir.ui.view">
```
Comment (dalonsod) [inline sale_order_type_quotation_number/views/sale_order_type_view.xml:5]: Could you also add `quotation_sequence_id` to `form` and `search` views?

**Review by dalonsod — CHANGES_REQUESTED** [review]
Tested right now in runboat:
* Created a new Sales Quotation Order sequence:
imagen
* Then assigned to Normal Order sales order type:
imagen
* Created a new sales quotation => didn't use expected quotation sequence:
imagen

Could you check?

**Comment by manuelregidor** [comment]
@dalonsod Please, make sure that the option marked in the following screenshot is unchecked. Thank you.

quotation_enumeration

**Comment by dalonsod** [comment]
Ok, tested again in runboat:

Set in sales order type a custom sales quotation sequence:

imagen

Defined as follows:

imagen

Then, creating a new Sales Order, is still using default SQ sequence, not custom one:

imagen

I hope I'm not doing something wrong again...

**Comment by manuelregidor** [comment]
@dalonsod Sorry, my fault. I forgot to create this PR https://github.com/OCA/sale-workflow/pull/2752 to improve abstraction so this module can work properly.

File: sale_order_type_quotation_number/views/sale_order_type_view.xml (hunk 1 above)
Comment (manuelregidor) [reply sale_order_type_quotation_number/views/sale_order_type_view.xml:5]: Done

**Comment by manuelregidor** [comment]
@dalonsod It should be working now. Thank you.

File: sale_order_type_quotation_number/views/sale_order_type_view.xml (hunk 1 above)
Comment (dalonsod) [reply sale_order_type_quotation_number/views/sale_order_type_view.xml:5]: Well, I was talking about adding them to `sale.order.type` model, not `sale.order`...

**Comment by dalonsod** [comment]
Tested in runboat :+1:

**Comment by HaraldPanten** [comment]
> Tested in runboat 👍

@dalonsod do you mind adding your review properly, approving this PR through "Review changes" option in "Files changed" tab? 🙏

**Comment by dalonsod** [comment]
I'm still waiting for a minor change in tree anda search views for sale.order.type, you can look at comments.

It's not blocking, if that minor improvement won't be done for now, please tell me and I'll make my approval

**Comment by manuelregidor** [comment]
@dalonsod We have identified a problem when using this module along with website sales. The thing is that it is necessary to apply the quotation sequence number set in the sale order type when the create method is called in sale.order. However, when a sale.order is created from a website sale, the sale order type is not available as a parameter at the create method, but it is calculated afterwards at the _compute_sale_type_id method. Our idea is to create a new module that makes website sales and this modules compatible. The module would add the possibility of selecting a default sale order type for website sales that would be applied at create method along with the quotation sequence selected in that sale order type. What do you think? Thank you.

**Comment by dalonsod** [comment]
I think it could be an excellent improvement, please call for help if needed. Are you thinking about linking `sale_order_type` and `website_sale`, without initial dependency with `sale_quotation_number`?

**Comment by manuelregidor** [comment]
@dalonsod Yes, that might be the best option. What do you think? @HaraldPanten @etobella @rousseldenis

**Comment by HaraldPanten** [comment]
Sounds good for me, but maybe other users can contribute with a better technical knowledge/opinion than mine.

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻 . I would indicate in the roadmap that it does not cover orders made from the website.

**Comment by ValentinVinagre** [comment]
@HaraldPanten could you do a review?

**Comment by HaraldPanten** [comment]
> @HaraldPanten could you do a review?

Waiting for these improvements --> https://github.com/OCA/sale-workflow/pull/2745#issuecomment-1787675623

File: sale_order_type_quotation_number/views/sale_order_type_view.xml (hunk 1 above)
Comment (ValentinVinagre) [reply sale_order_type_quotation_number/views/sale_order_type_view.xml:5]: @manuelregidor review please

Comment (manuelregidor) [reply sale_order_type_quotation_number/views/sale_order_type_view.xml:5]: @dalonsod Done

**Comment by manuelregidor** [comment]
@dalonsod Could you review, please? Thank you.

**Review by dalonsod — APPROVED** [review]
:+1:

**Review by HaraldPanten — APPROVED** [review]
LGTM

**Pull request merged**